### PR TITLE
Adding Microsoft DHCP Scope Monitoring

### PR DESCRIPTION
**Pull request by eg2965**
Hi Guys - This pull request is to add monitoring functionality in LibreNMS for Microsoft Windows DHCP scopes. I've tested it on my own Dev instance and it works well.

PLEASE NOTE: Even though I've only defined DHCP-MIB in the windows.yaml file, the MSFT-MIB file is required for this to function. I've tested it and it breaks when that file is missing.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by eg2965** [comment]
Adding screenshots of my Dev instance...

LNMS-Device-Health-Menu
LNMS-Edit-Health-Menu
LNMS-Health-Count-Menu

**Comment by eg2965** [comment]
Adding test data...

[msft_windows_svr_2019_1809.json](https://github.com/user-attachments/files/17878981/msft_windows_svr_2019_1809.json)

File: includes/definitions/discovery/windows.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+mib: DHCP-MIB
+modules:
+    os:
+        sysDescr_regex: '/Hardware:\s([A-Za-z0-9]+( [A-Za-z0-9]+)+)/([A-Za-z0-9]+( [A-Za-z0-9]+)+)\s-\sSoftware: Windows Version\s[0-9]+\.[0-9]+\s\(([A-Za-z0-9]+( [A-Za-z0-9]+)+)\)/'
```
Comment (Jellyfrog) [inline includes/definitions/discovery/windows.yaml:4]: This does not extract the fields. Need to use capture groups

File: mibs/dell/DHCP-MIB [hunk 2]
```diff
@@ -0,0 +1,167 @@
+          DHCP-MIB DEFINITIONS ::= BEGIN
```
Comment (Jellyfrog) [inline mibs/dell/DHCP-MIB:1]: Why is this in the dell folder ?

File: mibs/dell/MSFT-MIB [hunk 3]
```diff
@@ -0,0 +1,18 @@
+MSFT-MIB DEFINITIONS ::= BEGIN
+
```
Comment (Jellyfrog) [inline mibs/dell/MSFT-MIB:2]: Ditto

**Comment by Jellyfrog** [comment]
> Adding test data...
> 
> [msft_windows_svr_2019_1809.json](https://github.com/user-attachments/files/17878981/msft_windows_svr_2019_1809.json)


You should commit this file. And the snmprec file

File: includes/definitions/discovery/windows.yaml (hunk 1 above)
Comment (eg2965) [reply includes/definitions/discovery/windows.yaml:4]: @Jellyfrog - Is this something you can help me with? The regex identifier is the only component of this pull request that I'm not 100% comfortable with.

File: mibs/dell/MSFT-MIB (hunk 3 above)
Comment (eg2965) [reply mibs/dell/MSFT-MIB:2]: Same answer as above. I'm willing to go an alternate route with the folder location, but I'd like to get your thoughts first.

**Comment by eg2965** [comment]
> > Adding test data...
> > [msft_windows_svr_2019_1809.json](https://github.com/user-attachments/files/17878981/msft_windows_svr_2019_1809.json)
> 
> You should commit this file. And the snmprec file

Done! Just committed both files.

File: mibs/dell/DHCP-MIB (hunk 2 above)
Comment (murrant) [reply mibs/dell/DHCP-MIB:1]: Two options:
put them in a windows folder (new dir, no yaml changes needed)
put them in a microsoft folder and set `group: microsoft` in the definition yaml

File: includes/definitions/discovery/windows.yaml (hunk 1 above)
Comment (laf) [reply includes/definitions/discovery/windows.yaml:4]: I've run out of time tonight but we can optimise this regex. I'll try and get an example tomorrow.

**Review by laf — APPROVED** [review]
LGTM now. Thanks for the contribution.

**Comment by eg2965** [comment]
Thanks @laf! Appreciate you putting in the work to get this added.

**Pull request merged**